app/features/datasources/utils/passwordHandlers.ts

```diff
@@ -0,0 +1,15 @@
+export enum PasswordFieldEnum {Password = 'password', BasicAuthPassword = 'basicAuthPassword'}
+export type Ctrl = {
+  current: {
+    secureJsonFields: {
+      [key: string]: boolean;
+    };
+    secureJsonData?: {
+      [key: string]: string;
+    };
+    password?: string;
+    basicAuthPassword?: string;
+  };
+};
+export const createResetHandler = (ctrl: Ctrl, field: PasswordFieldEnum) => (event: any) => {};
```

Comment:
```suggestion
export declare function createResetHandler(ctrl: Ctrl, field: PasswordFieldEnum): (event: any) => void;
```

Reply:
Will do but curious which difference it makes? Thank you!

Reply:
When you use `const`, you create a variable. When you use `declare`, you tell the TypeScript that variable exists.

https://stackoverflow.com/a/40700848